src/adagrams.js

```diff
@@ -1,8 +1,96 @@
+const _ = require('lodash')
+
+const letterPool = {
+  'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12, 'F': 2, 'G': 3,
+  'H': 2, 'I': 9, 'J': 1, 'K': 1, 'L': 4, 'M': 2, 'N': 6,
+  'O': 8, 'P': 2, 'Q': 1, 'R': 6, 'S': 4, 'T': 6, 'U': 4,
+  'V': 2, 'W': 2, 'X': 1, 'Y': 2, 'Z': 1,
+}
+
 const Adagrams = {
   drawLetters() {
-    // Implement this method for wave 1
+    const keys = Object.keys(letterPool);
+    const length = keys.length;
+    const lettersInHand = [];
+
+    for (let i = 0;i < 10;i += 1) {
+      const randomIndex = Math.floor(Math.random() * length);
```

Comment:
All letters are equally likely in the scenario you set up, a hash-like-object won't preserve the disparity in quantity. Better to convert the object into an array.